src/clientAtom.ts

```diff
@@ -1,13 +1,23 @@
-import { InMemoryCache, ApolloClient } from '@apollo/client'
+import {
+  InMemoryCache,
+  ApolloClient,
+  NormalizedCacheObject,
+} from '@apollo/client'
 import { atom } from 'jotai/vanilla'
 
 const DEFAULT_URL =
   (typeof process === 'object' && process.env.JOTAI_APOLLO_DEFAULT_URL) ||
   '/graphql'
 
-const client = new ApolloClient({
-  uri: DEFAULT_URL,
-  cache: new InMemoryCache(),
-})
+let defaultClient: ApolloClient<NormalizedCacheObject> | null = null
+
+export const clientAtom = atom(() => {
```

Comment:
Unfortunately, this make the atom read-only, which prevents users to use custom client.